### PR TITLE
LinearTrendTransform bug fix

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (checklist)

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `make lint` (`poetry install -E style`).
- [ ] Did you make sure to update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you check that your code passes the unit tests `pytest tests/` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes

  -
  -
  -

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Closing issues

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: etna/transforms/detrend.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def fit(self, df: pd.DataFrame) -> "_OneSegmentLinearTrendBaseTransform":
         _OneSegmentLinearTrendBaseTransform
             instance with trained regressor
         """
+        df = df.dropna()
```
Comment (martins0n) [inline etna/transforms/detrend.py:47]: What if `None` in the middle of df?
We could use something like this https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.first_valid_index.html

**Review by martins0n — CHANGES_REQUESTED** [review]
We should add test case for this bug

File: tests/test_transforms/test_detrend_transform.py [hunk 2]
```diff
@@ -189,3 +196,35 @@ def test_inverse_transform_theil_sen_trend_two_segments(df_two_segments: pd.Data
     """
     trend_transform = TheilSenTrendTransform(in_column="target", n_subsamples=len(df_two_segments))
     _test_inverse_transform_many_segments(trend_transform=trend_transform, df=df_two_segments)
+
+
+def test_fit_transform_linear_trend_two_segments_diff_size(df_two_segments_diff_size: pd.DataFrame):
```
Comment (martins0n) [inline tests/test_transforms/test_detrend_transform.py:201]: For repeated test cases it's better to use https://docs.pytest.org/en/6.2.x/parametrize.html, you could pass transform as parameter to test case.

**Review by martins0n — CHANGES_REQUESTED** [review]

File: tests/test_transforms/test_detrend_transform.py (hunk 2 above)
Comment (alex-hse-repository) [reply tests/test_transforms/test_detrend_transform.py]: Done

**Review by martins0n — APPROVED** [review]
👍

**Pull request merged**